src/basic/env-util.c

```diff
@@ -784,3 +785,42 @@ int setenv_systemd_exec_pid(bool update_only) {
 
         return 1;
 }
+
+int getenv_path_list(const char *name, char ***ret_paths) {
+        _cleanup_strv_free_ char **l = NULL;
+        const char *e;
+        char **p;
+        int r;
+
+        assert(name);
+        assert(ret_paths);
+
+        e = secure_getenv(name);
+        if (!e)
+                return 0;
+
+        r = strv_split_full(&l, e, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse $%s: %m", name);
+
+        STRV_FOREACH(p, l) {
+                if (!path_is_absolute(*p))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Path '%s' is not absolute, refusing.", *p);
+
+                if (!path_is_normalized(*p))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Path '%s' is not normalized, refusing.", *p);
+
+                if (path_equal(*p, "/"))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Path '%s' is the root fs, refusing.", *p);
+        }
+
+        if (strv_isempty(l))
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "No paths specified, refusing.");
+
+        strv_free_and_replace(*ret_paths, l);
```

Comment:
hmm, so this is a function now that has a parameter `ret_paths` that suggests it is a return parameter, but actually it's both input *and* output now. Which is icky.

At the very least this param should be renamed (to drop the ret_) prefix and get a comment in the param list saying `/* input + output! */` or so...

but quite frankly I'd just make it output-only, i.e. make the caller free the old array on their own.

Reply:
yeah I had just moved over the existing function, which had strv_free_and_replace, and didn't really think twice about it. But yes it doesn't make much sense, so I've fixed as suggested.